src/sid/update_states.py

```diff
@@ -166,3 +163,66 @@ def _kill_people_over_icu_limit(states, params, seed):
         states.loc[to_kill, "cd_dead_true"] = 0
 
     return states
+
+
+def _compute_new_tests_with_share_known_cases(
+    states: pd.DataFrame, share_known_cases: float
+) -> pd.Series:
+    # Get all active cases.
+    is_active_case = states.eval(IS_ACTIVE_CASE)
+    n_active_cases = is_active_case.sum()
+
+    # Identify active and known cases. We treat individuals whose test is processed
+    # as known cases, to not assign to many tests.
+    is_potentially_known_case = (is_active_case & states["knows_immune"]) | (
+        states["cd_received_test_result_true"] > 0
+    )
+    n_additional_known_and_active_cases = int(
+        n_active_cases * share_known_cases - is_potentially_known_case.sum()
+    )
+
+    if n_additional_known_and_active_cases > 0:
+        ilocs = np.arange(len(states))[is_active_case & ~is_potentially_known_case]
```

Comment:
Why go over `np.arange` here and not just sample indices directly, i.e. `states[is_active_case & ~is_potentially_known_case].index` ?

Reply:
`ilocs` are better thinking about the general case where the index could include non-numeric values or is a multiindex. Of course, this does not make a difference here. Maybe ilocs are faster?

I think since both operations are equivalent and I have used ilocs in other places as well, I would leave it like this.